include/circt/Dialect/RTG/IR/RTGOps.td

```diff
@@ -95,3 +95,44 @@ def InvokeSequenceOp : RTGOp<"invoke_sequence", []> {
 
   let assemblyFormat = "$sequence attr-dict";
 }
+
+//===- Set Operations ------------------------------------------------------===//
+
+def SetCreateOp : RTGOp<"set_create", [Pure, SameTypeOperands]> {
+  let summary = "constructs a set of the given values";
+
+  let arguments = (ins Variadic<AnyType>:$elements);
+  let results = (outs SetType:$set);
+
+  let hasCustomAssemblyFormat = 1;
+  let hasVerifier = 1;
+}
+
+def SetSelectRandomOp : RTGOp<"set_select_random", [
+  Pure,
+  TypesMatchWith<"output must be of the element type of input set", 
+                 "set", "output",
+                 "llvm::cast<rtg::SetType>($_self).getElementType()">
+]> {
+  let summary = "selects an element uniformly at random from a set";
```

Comment:
What does this return if the set is empty?

Reply:
This is currently undefined behavior. My plan is to add another operation to query the number of elements in the set. `scf.if` could then be used to skip random selections from empty sets. Alternatively, we could pass a default value to be chosen if the set is empty. Feel free to share your thoughts/opinion on how this should ideally be handled.